addons/hr_evaluation/hr_evaluation.py

```diff
@@ -1,375 +1,465 @@
+    email_to = fields.Char(string='Appraisal Receiver')
+    appraisal_url = fields.Char(string='Appraisal URL')
+    tot_sent_appraisal = fields.Integer(string='Number of sent appraisal', compute='_get_tot_sent_appraisal', default=0, store=True)
+    tot_comp_appraisal = fields.Integer(string='Number of completed appraisal', compute="_get_tot_completed_appraisal")
+    user_id = fields.Many2one('res.users', string='Related User', default=lambda self: self.env.uid)
+
+    @api.one
+    @api.depends('employee_id')
+    def _set_display_name(self):
+        self.display_name = self.employee_id.name_related
+
+    @api.onchange('employee_id')
+    def onchange_employee_id(self):
+        if self.employee_id:
+            self.department_id = self.employee_id.department_id
+            self.appraisal_manager = self.employee_id.appraisal_manager
+            self.appraisal_manager_ids = self.employee_id.appraisal_manager_ids
+            self.appraisal_manager_survey_id = self.employee_id.appraisal_manager_survey_id
+            self.appraisal_colleagues = self.employee_id.appraisal_colleagues
+            self.appraisal_colleagues_ids = self.employee_id.appraisal_colleagues_ids
+            self.appraisal_colleagues_survey_id = self.employee_id.appraisal_colleagues_survey_id
+            self.appraisal_self = self.employee_id.appraisal_self
+            self.appraisal_employee = self.employee_id.appraisal_employee or self.employee_id.name
+            self.appraisal_self_survey_id = self.employee_id.appraisal_self_survey_id
+            self.appraisal_subordinates = self.employee_id.appraisal_subordinates
+            self.appraisal_subordinates_ids = self.employee_id.appraisal_subordinates_ids
+            self.appraisal_subordinates_survey_id = self.employee_id.appraisal_subordinates_survey_id
+
+    @api.one
+    @api.constrains('employee_id', 'department_id', 'date_close')
+    def _check_employee_appraisal_duplication(self):
+        """ Avoid duplication"""
+        if self.employee_id and self.department_id and self.date_close:
+            date_closed = datetime.datetime.strptime(self.date_close, DTF)
+            start_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-01' + ' 00:00:00')
+            end_datetime = time.strftime(str(date_closed.year) + '-' + str(date_closed.month) + '-' + str(date_closed.day) + ' 23:59:59')
+            appraisal_ids = self.search([
+                ('employee_id', '=', self.employee_id.id), ('department_id', '=', self.department_id.id),
+                ('date_close', '<=', end_datetime),
+                ('date_close', '>=', start_datetime)])
+            if len(appraisal_ids) > 1:
+                raise UserError(_("You cannot create more than one appraisal for same Month & Year"))
+
+    @api.model
+    def create_message_subscribe_users(self, subscribe_users):
```

Comment:
Why do you have a subscribe_users ina ddition to other users ? You should compute all subscribers in this method, based on self. And it is obviously not a model method.